src/app/(app)/create-todo.tsx

```diff
@@ -0,0 +1,61 @@
+import DateSelect from "@/components/DateSelect";
+import Header from "@/components/Header";
+import ScreenView from "@/components/ScreenView";
+import TextButton from "@/components/TextButton";
+import TextInput from "@/components/TextInput";
+import { useUser } from "@/hooks/useUser";
+import { db, id } from "@/utilities/database";
+import { useRouter } from "expo-router";
+import { generateKeyBetween } from "fractional-indexing";
+import { useState } from "react";
+import { View } from "react-native";
+
+export default function CreateTodo() {
+  const [label, setLabel] = useState<string>("");
+  const [date, setDate] = useState<string>("");
+  const user = useUser();
+  const router = useRouter();
+
+  const pushTodo = async () => {
+    const { data } = await db.queryOnce({
```

Comment:
One thing with this: 

This will slow down the 'create' flow a bit. 

I _think_ you may already have the last todo's `position` when you are in the `CreateTodo` state. Is there a way you can pass that in to `CreateTodo`, so there would be no need to wait here?